### PR TITLE
add mode to stats:poisson_conf_limits

### PR DESCRIPTION
**Pull request by hamogu**
'kraft-burrows-nousek'** This is a Bayesian appraoch which allows
    for the presence of a known background :math:`B` in the source signal
    :math:`N`.

http://adsabs.harvard.edu/abs/1991ApJ...374..344K

### CODE REVIEW

**Comment by pllim** [comment]
Appveyor failed in Python 3.4
```python
        elif interval == 'kraft-burrows-nousek': 
            if conflevel is None: 
                raise ValueError('Set conflevel for method {0}. (sigma is ignored.)'.format(interval)) 
            conflevel = np.asanyarray(conflevel) 
            if np.any(conflevel <=0) or np.any(conflevel >= 1): 
                raise ValueError('Conflevel must be a number between 0 and 1.') 
>           if background < 0: 
E           TypeError: unorderable types: list() < int() 
 
astropy\stats\funcs.py:698: TypeError 
= 1 failed, 7464 passed, 96 skipped, 69 xfailed, 3 xpassed i
```

File: astropy/stats/funcs.py [hunk 1]
```diff
@@ -444,19 +446,35 @@ def true_efficiency(x):
...
         Number of counts (0 <= ``n``).
-    interval : {'root-n','root-n-0','pearson','sherpagehrels','frequentist-confidence'}, optional
+    interval : {'root-n','root-n-0','pearson','sherpagehrels','frequentist-confidence', 'kraft-burrows-nousek'}, optional
         Formula used for confidence interval. See notes for details.
         Default is ``'root-n'``.
     sigma : float
         Number of sigma for confidence interval; only supported for
         the 'frequentist-confidence' mode.
+    background : float
+        Number of counts exspected from the background; only supported for
```
Comment (mirca) [inline astropy/stats/funcs.py]: expected*

File: astropy/stats/funcs.py [hunk 2]
```diff
@@ -638,6 +689,17 @@ def poisson_conf_interval(n, interval='root-n', sigma=1):
...
             conf_interval[0,n==0] = 0
+    elif interval == 'kraft-burrows-nousek':
+        if conflevel is None:
+            raise ValueError('Set conflevel for method {0}. (sigma is ignored.)'.format(interval))
+        conflevel = np.asanyarray(conflevel)
+        if np.any(conflevel <=0) or np.any(conflevel >= 1):
+            raise ValueError('Conflevel must be a number between 0 and 1.')
+        background = np.asanyarray(background)
+        if np.any(background < 0):
+            raise ValueError('Background must be >= 1.')
```
Comment (mirca) [inline astropy/stats/funcs.py]: Sorry if I am mistaken, but, after reading the paper, I couldn't find a reason to explain why ``Background must be >= 1.``. Also, the code is checking for ``np.any(background < 0)``.

Comment (hamogu) [inline astropy/stats/funcs.py]: Thanks. That's just a typo. The code is correct. Will fix soon. (I just wanted a check here to catch people who accidentally put in a negative background or a complex number or a string.)

File: astropy/stats/funcs.py [hunk 3]
```diff
@@ -444,19 +446,35 @@ def true_efficiency(x):
...
+    interval : {'root-n','root-n-0','pearson','sherpagehrels','frequentist-confidence', 'kraft-burrows-nousek'}, optional
         Formula used for confidence interval. See notes for details.
         Default is ``'root-n'``.
     sigma : float
         Number of sigma for confidence interval; only supported for
         the 'frequentist-confidence' mode.
+    background : float
+        Number of counts exspected from the background; only supported for
+        the 'kraft-burrows-nousek' mode. This number is assumed to be determined
+        from a large region so that the uncertaintiy on its value is negligible.
```
Comment (mirca) [inline astropy/stats/funcs.py]: uncertainty*

File: astropy/stats/funcs.py [hunk 4]
```diff
@@ -530,6 +548,36 @@ def poisson_conf_interval(n, interval='root-n', sigma=1):
     distribution (at the point given by the parameter 'sigma'). See
     [Maxwell 2011][maxw11] for further details.
 
+    **6. 'kraft-burrows-nousek'** This is a Bayesian appraoch which allows
```
Comment (mirca) [inline astropy/stats/funcs.py]: approach*

**Comment by pllim** [comment]
Now some Travis tests failed
```python
=================================== FAILURES ===================================

______________________ test_poisson_limit_nodependencies _______________________

    @pytest.mark.skipif('HAS_SCIPY or HAS_MPMATH')

    def test_poisson_limit_nodependencies():

        with pytest.raises(ImportError) as e:

>           a = funcs.poisson_conf_interval(20., 10., .95)
```

**Comment by mirca** [comment]
Although this problem seems insensitive for the choice of a prior, I think it should be stated somewhere that the code uses a positively defined uniform prior. (The authors also use exponential and Lorentzian priors, for comparison).

**Comment by hamogu** [comment]
I've added only a single line that's not covered by tests and that line is for compatibility with numpy 1.6 (where we don't run coveralls).
So, I request to ignore the coveralls failure here.

**Comment by hamogu** [comment]
@astrofrog : Can you restart the Travis tests. Seems that continuum.io was down (or things happened in ci-helpers ...)

**Comment by astrofrog** [comment]
See @cdeil's message on astropy-dev. I think this isn't fixed yet, but will try.

**Comment by hamogu** [comment]
I get the astropy-dev in "daily digest" so I did not see that yet.
Thanks! Will wait until continuum's servers are up again.

**Comment by hamogu** [comment]
Any more comments anyone or can this be merged?

**Comment by astrofrog** [comment]
@crawfordsm @astronomeralex - any comments on this (as astropy.stats maintainers), or can we merge it?

**Comment by crawfordsm** [comment]
Finally got a chance to give this a final-review.   Thanks for the contribution @hamogu !

**Pull request merged**